### PR TITLE
PF-1184: Add new reference type in the CLI command: git repo

### PR DESCRIPTION
**Pull request by yuhuyoyo**
Add CLI command for create, update, delete a git repo reference in the workspace. 
e.g. terra resource add-ref git-repo --name=wsm-repo-reference --repo-url=https://github.com/DataBiosphere/terra-workspace-manager.git

Because we don't have a way to store credential in user profile in the workspace yet, we currently don't check access to the git repo when creating the reference.

### CODE REVIEW

File: src/main/java/bio/terra/cli/businessobject/resource/GitRepo.java [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+   */
+  public static GitRepo addReferenced(AddGitRepoParams addGitRepoParams) {
+    validateEnvironmentVariableName(addGitRepoParams.resourceFields.name);
+
+    // call WSM to add the reference. use the pet SA credentials instead of the end user's
+    // credentials, because they include the cloud-platform scope. WSM needs the cloud-platform
+    // scope to perform its access check before adding the reference. note that this means a user
+    // cannot add a reference unless their pet SA has access to it.
+    GitRepoResource addedResource =
+        WorkspaceManagerService.fromContextForPetSa()
```
Comment (marikomedlock) [inline src/main/java/bio/terra/cli/businessobject/resource/GitRepo.java:69]: Does this apply to this reference type? The PR description indicates you're not checking access.

File: src/main/java/bio/terra/cli/businessobject/resource/GitRepo.java [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+  /** Delete a Git repo referenced resource in the workspace. */
+  protected void deleteReferenced() {
+    // call WSM to delete the reference
+    WorkspaceManagerService.fromContext()
+        .deleteReferencedGitRepo(Context.requireWorkspace().getId(), id);
+  }
+
+  @Override
+  protected void deleteControlled() {
+    logger.warn("Controlled git repo resource is not supported in workspace.");
```
Comment (marikomedlock) [inline src/main/java/bio/terra/cli/businessobject/resource/GitRepo.java:99]: Please throw an `UnsupportedOperationException` here instead.

File: src/main/java/bio/terra/cli/service/WorkspaceManagerService.java [hunk 3]
```diff
@@ -652,6 +642,34 @@ public GcpBigQueryDatasetResource createReferencedBigQueryDataset(
...
+      UUID workspaceId, AddGitRepoParams addGitRepoParams) {
+    CreateGitRepoReferenceRequestBody createGitRepoReferenceRequestBody =
+        new CreateGitRepoReferenceRequestBody()
+            .metadata(getReferencedResourceMetadata(addGitRepoParams.resourceFields))
+            .gitrepo(new GitRepoAttributes().gitRepoUrl(addGitRepoParams.gitRepoUrl));
+    return callWithRetries(
+        () ->
+            new ReferencedGcpResourceApi(apiClient)
+                .createGitRepoReference(createGitRepoReferenceRequestBody, workspaceId),
+        "Error when creating referenced git repo referenced resource in the workspace");
```
Comment (marikomedlock) [inline src/main/java/bio/terra/cli/service/WorkspaceManagerService.java:670]: nit: "referenced" twice

File: src/test/java/unit/GitRepoReferenced.java [hunk 4]
```diff
@@ -0,0 +1,351 @@
...
+    assertEquals(name, matchedResource.name, "list output matches name");
+    assertEquals(
+        GIT_REPO_HTTPS_URL, matchedResource.gitRepoUrl, "list output matches git repo url");
+
+    // `terra resource describe --name=$name --format=json`
+    UFGitRepo describeResource =
+        TestCommand.runAndParseCommandExpectSuccess(
+            UFGitRepo.class, "resource", "describe", "--name=" + name);
+
+    // check that the url and git repo url match
```
Comment (marikomedlock) [inline src/test/java/unit/GitRepoReferenced.java:67]: nit: name and git repo url

File: src/test/java/unit/GitRepoReferenced.java [hunk 5]
```diff
@@ -0,0 +1,351 @@
...
+    assertEquals(yetAnotherName, describeGitRepoAfterUpdatingReferenceTarget.name);
+  }
+
+  /**
+   * Helper method to call `terra resources list` and filter the results on the specified resource
+   * name and workspace (uses the current workspace if null).
+   */
+  static List<UFGitRepo> listGitRepoResourcesWithName(String resourceName)
+      throws JsonProcessingException {
+    // `terra resources list --type=GCS_OBJECT --format=json`
```
Comment (marikomedlock) [inline src/test/java/unit/GitRepoReferenced.java:341]: nit: type=GIT_REPO

**Review by marikomedlock — COMMENTED** [review]
This looks great, thanks!

One question about whether we need to call WSM with the pet SA token.

**Review by zloery — APPROVED** [review]
LGTM with one comment.

Could you also update the README with the new referenced resource type? thanks!

File: src/main/java/bio/terra/cli/businessobject/resource/GitRepo.java [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+   *
+   * @return the resource that was added
+   */
+  public static GitRepo addReferenced(AddGitRepoParams addGitRepoParams) {
+    validateEnvironmentVariableName(addGitRepoParams.resourceFields.name);
+
+    // call WSM to add the reference. use the pet SA credentials instead of the end user's
+    // credentials, because they include the cloud-platform scope. WSM needs the cloud-platform
+    // scope to perform its access check before adding the reference. note that this means a user
+    // cannot add a reference unless their pet SA has access to it.
```
Comment (marikomedlock) [inline src/main/java/bio/terra/cli/businessobject/resource/GitRepo.java:67]: remove comment, out-of-date

**Review by marikomedlock — APPROVED** [review]

**Pull request merged**